src/swaps/boltz.rs

```diff
@@ -310,16 +310,42 @@ impl Fees {
     }
 }
 
+/// States for a submarine swap.
+///
+/// See <https://docs.boltz.exchange/v/api/lifecycle#normal-submarine-swaps>
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(rename_all = "snake_case")]
 pub enum SubSwapStates {
+    /// Initial state of the swap; optionally the initial state can also be `invoice.set` in case
+    /// the invoice was already specified in the `/createswap` request.
     Created,
+    /// The lockup transaction is found in the mempool, meaning the user is sending funds to the
+    /// lockup chain address.
     TransactionMempool,
+    /// The lockup transaction was included in a block.
     TransactionConfirmed,
+    /// If the invoice was _not_ set as part of the `/createswap` request, this state confirms that
+    /// an invoice with the correct amount and hash was set.
     InvoiceSet,
+    /// Boltz successfully paid the invoice.
     InvoicePaid,
+    /// Boltz started paying the invoice.
+    InvoicePending,
+    /// Boltz failed to pay the invoice. In this case the user needs to broadcast a refund
+    /// transaction to reclaim the locked up chain L-BTC.
```

Comment:
```suggestion
    /// transaction to reclaim the locked up onchain coins.
```

Could also be on the mainchain